b2b_ecommerce/views.py

```diff
@@ -104,6 +106,10 @@ def post(self, request, *args, **kwargs):  # pylint: disable=too-many-locals
             url = settings.CYBERSOURCE_SECURE_ACCEPTANCE_URL
             method = "POST"
 
+        if not User.objects.filter(email=email).exists():
+            sync_hubspot_b2b_contact(email)
```

Comment:
This should work but it seems that on hubspot the deal is often synced before contact. I tried this change to `sync_hubspot_b2b_deal` to delay its execution by 2 minutes and the contact was successfully linked to the deal:

```python
tasks.sync_b2b_deal_with_hubspot.apply_async(
    countdown=120, 
    kwargs={"order_id": order.id}
)
```